docs/dev/debugging.rst

```diff
@@ -0,0 +1,51 @@
+Debugging
+=========
+
+
+.. _dev-debugging-kernel-oops:
+
+Kernel Oops
+-----------
+
+Sometimes a running Linux kernel detects an error during runtime that canot
+be corrected.
+This usually generates a stack trace that points to the location in the code
+that caused the oops.
+
+Linux kernels in Gluon (and OpenWrt) are stripped.
+That means they do not contain any debug symbols.
+On one hand this leads to a smaller binary and faster loading times on the
+target.
+On the other hand this means that in a case of a stack trace the unwinder
+can only print memory locations and no further debugging information.
+
+Gluon stores a compressed kernel with debug symbols for every architecture
+in the directory `output/debug/`.
+These kernels should be kept along with the images as long as the images
+are in use.
+This allows the developer to analyse a stack trace later.
+
+Decoding Stacktraces
+....................
+
+The tooling is contained in the kernel source tree in the file
+`decode_stacktrace.sh <https://github.com/torvalds/linux/blob/master/scripts/decode_stacktrace.sh>`__.
+This file and the needed source tree are available in the directory: ::
+
+    openwrt/build_dir/target-<architecture>/linux-<architecture>/linux-<version>/
+
+.. note::
+   Make sure to use a kernel tree that matches the version and patches
+   that was used to build the kernel.
+   If in doubt just re-build the images for the architecture.
+
+Some more information on how to use this tool can be found at
+`LWN <https://lwn.net/Articles/592724/>`__.
+
+Obtaining Stacktraces on ar71xx
```

Comment:
AFAIK crashlog is not dedicated to ar71xx but it is disabled on some architectures/targets.

https://github.com/openwrt/openwrt/blob/openwrt-19.07/config/Config-kernel.in#L31

Reply:
With all those exclusions, that leaves us with... mips(el), I guess? Although it seems this is additionally enabled `target/linux/generic/config-4.14`, but I have no idea if the disabled OpenWrt .config setting has precedence over this defconfig file.

Reply:
I would suggest to leave it with ar71xx for now? The hint where it may be stored is still in the docs. And if anyone wants to add other targets/architectures they are free to do so anytime.

Reply:
At least in the kernel .config, the crashlog is enabled for all targets except for x86 - the OpenWrt Kconfig symbols only limits on which archs it is possible to override this via OpenWrt's .config. This seems like a bug to me, but to fix it, we should first determine on which archs the crashlog actually works... Also note that in OpenWrt master, crashlog support hasn't been ported to Kernel 5.4 yet.

For this doc file, I would just remove the target from the title, and write something like "On many targets, ..." in the text.

Reply:
Done. Documentation now reads:
```
Obtaining Stacktraces
.....................

On many targets stacktraces can be read from the following
location after reboot: ::

    /sys/kernel/debug/crashlog

```